code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -750,6 +750,111 @@
 	var/obj/item/pushbroom/cyborg/BR = locate() in R.model.modules
 	if (BR)
 		R.model.remove_module(BR, TRUE)
+		
+/obj/item/borg/upgrade/condimentsynthesizer
+	name = "Service Cyborg Condiment Synthesiser"
+	desc = "An upgrade to the service model cyborg, allowing it \
+		to produce solid condiments."
+	icon_state = "cyborg_upgrade3"
+	require_model = TRUE
+	model_type = list(/obj/item/robot_model/service)
+	model_flags = BORG_MODEL_SERVICE
+
+/obj/item/borg/upgrade/condimentsynthesizer/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/reagent_containers/borghypo/condimentsynthesizer/CS = locate() in R.model.modules
+		if(CS)
+			to_chat(user, span_warning("This unit has no room for another condiment synthesizer!"))
+			return FALSE
+		CS = new(R.model)
+		R.model.basic_modules += CS
+		R.model.add_module(CS, FALSE, TRUE)
+
+/obj/item/borg/upgrade/condimentsynthesizer/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/reagent_containers/borghypo/condimentsynthesizer/CS = locate() in R.model.modules
+		if (CS)
+			R.model.remove_module(CS, TRUE)
+			
+/obj/item/borg/upgrade/siliconknife
+	name = "Service Cyborg Kitchen Toolset"
+	desc = "An upgrade to the service model cyborg, to help process foods"
+	icon_state = "cyborg_upgrade3"
+	require_model = TRUE
+	model_type = list(/obj/item/robot_model/service)
+	model_flags = BORG_MODEL_SERVICE
+			
+/obj/item/borg/upgrade/siliconknife/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/knife/kitchen/silicon/SK = locate() in R.model.modules
+		if(SK)
+			to_chat(user, span_warning("This unit has no room for another Kitchen toolset!"))
```

Comment:
```suggestion
			to_chat(user, span_warning("This unit has no room for another [src]!"))
```
I would test the above though, my spidey senses are telling me that because the name starts with a capital letter it might treat it grammatically as a proper noun.